main.go

```diff
@@ -435,19 +437,17 @@ func main() {
 		}()
 	}
 
-	var registerClient = k8s.AdmissionWebhookRegisterClient{
-		Logger: logger.Named("admissionWebhookRegisterClient"),
-	}
+	var source *workloadapi.X509Source
 
-	err = registerClient.Register(ctx, conf)
-	if err != nil {
-		prod.Fatal(err.Error())
+	if strings.EqualFold(conf.WebhookMode, config.ModeAuto) {
+		registerClient := registerClient(ctx, conf, logger)
+		defer func() {
+			_ = registerClient.Unregister(context.Background(), conf)
+		}()
+	} else {
+		source = getX509Source(ctx, logger)
```

Comment:
You can move `source` and `getX509Source` directly into `prepareTLSConfig`.

Reply:
Done